SharpPcap/LibPcap/PcapDevice.cs

```diff
@@ -321,6 +320,58 @@ public virtual int GetNextPacket(out RawCapture p)
             return res;
         }
 
+        /// <summary>
+        /// Higher performance version of RawCapture GetNextPacket() through the use of Span<byte>
+        /// </summary>
+        /// <param name="handler"></param>
+        /// <returns>
+        /// A <see cref="int"/> that contains the result code. 1 upon success, 0 upon error.
+        /// </returns>
+        public virtual int GetNextPacket(PacketArrivalEventHandler handler)
```

Comment:
I think the API would look nicer like this
```suggestion
        public virtual int GetNextPacket(out CaptureEventArgs args)
```

Reply:
Yeah that's a good idea. Let me see if that's a valid change.